packages/console/src/pages/SignInExperience/utilities.ts

```diff
@@ -98,3 +100,51 @@ export const compareSignInMethods = (
 
   return Object.values(SignInMethodKey).every((key) => beforeMethods[key] === afterMethods[key]);
 };
+
+export const flattenObject = (
+  object: Record<string, unknown>,
+  keyPrefix = ''
+): Record<string, unknown> => {
+  return Object.keys(object).reduce((result, key) => {
+    const prefix = keyPrefix ? `${keyPrefix}.` : keyPrefix;
+    const dataKey = `${prefix}${key}`;
+    const data = object[key];
+
+    const temporaryObject =
+      typeof data === 'object'
+        ? // eslint-disable-next-line no-restricted-syntax
+          flattenObject(data as Record<string, unknown>, dataKey)
+        : { [dataKey]: data };
+
+    return {
+      ...result,
+      ...temporaryObject,
+    };
+  }, {});
+};
+
+const emptyTranslation = (translation: Record<string, unknown>): Record<string, unknown> => {
+  return Object.entries(translation).reduce<Record<string, unknown>>((result, [key, value]) => {
+    if (typeof value === 'object' && !Array.isArray(value)) {
+      return {
+        ...result,
+        // eslint-disable-next-line no-restricted-syntax
+        [key]: emptyTranslation(value as Record<string, unknown>),
+      };
+    }
+
+    if (typeof value === 'string') {
+      return {
+        ...result,
+        [key]: '',
+      };
+    }
+
+    return { ...result, [key]: value };
+  }, {});
+};
+
+export const createEmptyUiTranslation = () => {
+  // eslint-disable-next-line no-restricted-syntax
+  return emptyTranslation(en.translation) as UiTranslation;
+};
```

Comment:
Why create this method to call `emptyTranslation`?

Reply:
Because we empty the `en.translation` to get an empty `uiTranslation`.

Reply:
But it seems we should only use `en` as the default translation, so it's unnecessary to pass `en` as a prop...

Reply:
Also it would be better if we can use zod to guard return value type

Reply:
- The `en` is needed for this is a recursive function.
- It seems impossible to guard the `UiTranslation` by zod. Reference: https://github.com/colinhacks/zod/issues/53

Reply:
Add an issue to track this: https://linear.app/silverhand/issue/LOG-4235/handle-type-assertion-gracefully-in-sie-utilities